client/src/components/Dashboard/DashboardForecast/DashboardForecast.tsx

```diff
@@ -69,51 +56,39 @@ const DashboardForecast: FunctionComponent = () => {
     }
   };
 
-  const handleGetLocationWeather = useCallback(() => {
-    if (!query) return;
-
-    switch (selectedForecastType) {
-      case ForecastType.hourly:
-      default:
-        dispatch(getLocationHourlyWeather(query));
-        break;
-      case ForecastType.threeHourly: {
-        dispatch(getLocationThreeHourlyWeather(query));
-        break;
-      }
-      case ForecastType.daily: {
-        dispatch(getLocationDailyWeather(query));
-        break;
-      }
-    }
-  }, [dispatch, query, selectedForecastType]);
-
-  // Get data
   useEffect(() => {
-    handleGetLocationWeather();
-  }, [handleGetLocationWeather]);
+    if (isWeatherForecast) {
+      setSelectedForecastType(ForecastType.weatherHourly);
+    } else {
+      setSelectedForecastType(ForecastType.airQualityHourly);
+    }
+  }, [isWeatherForecast]);
 
   return (
     <S.DashboardForecast>
       <S.DashboardToolbar>
         <Flex justifySpaceBetween>
           <S.DashboardForecastTypes role="tablist">
             <Flex justifySpaceBetween>
-              {(Object.keys(ForecastType) as Array<keyof typeof ForecastType>).map(key => (
-                <S.DashboardForecastType
-                  tabIndex={0}
-                  role="tab"
-                  aria-controls="tabpanel"
-                  aria-selected={selectedForecastType === key}
-                  key={key}
-                  themeType={theme}
-                  active={selectedForecastType === key}
-                  onClick={() => handleForecastTypeOnClick(key)}
-                  onKeyPress={e => handleForecastTypeOnKeyPress(e, key)}
-                >
-                  {capitalize(key)}
-                </S.DashboardForecastType>
-              ))}
+              {(Object.keys(ForecastType) as Array<keyof typeof ForecastType>)
+                .filter(key =>
+                  isWeatherForecast ? key.startsWith('weather') : key.startsWith('air')
+                )
```

Comment:
```
export const filterThatStartsWith = <T extends string>(arr: T[], startsWith: string) =>
  arr.filter(key => key.startsWith(startsWith));

 {filterThatStartsWith<ForecastType>(
                Object.keys(ForecastType) as ForecastType[],
                isWeatherForecast ? 'weather' : 'air'
              ).map...
```
1. How about this?

2. Do I really need filtering to be a separate function if I am not reusing it? Is `.filter(key => key.startsWith(isWeatherForecast ? 'weather' : 'air'))` not enough?

Reply:
looks good, but you don't need to pass array as param, you can use `Object.keys(ForecastType)` because it's hardcoded

Reply:
I thought you meant creating a pure utility function.

Reply:
it's not a shared utility function, so it's ok for it not to be a pure one :) and about point 2 from previous comment: this code snippet contains only one dynamic variable, so i think it would be better to move it from rendered jsx.
P.S. actually, i forgot to suggest even better option: you can create 2 enums - one for weather types, another for air quality, and then just select between them like this:
`{Object.keys(isWeatherForecast ? WeatherType : AirQualityType).map()}`

Reply:
> P.S. actually, i forgot to suggest even better option: you can create 2 enums - one for weather types, another for air quality, and then just select between them like this:
{Object.keys(isWeatherForecast ? WeatherType : AirQualityType).map()}

Nice, thanks. I actually wanted to have two enums:)